docker/storm/templates/Dockerfile.template

```diff
@@ -0,0 +1,122 @@
+
+# Set default version to Python 3
+RUN update-alternatives --install "/usr/bin/python" "python" "$(which python3)" 1
+
+ARG GPG_KEY=51379DA8A7AE5B02674EF15C134716AF768D9B6E
+ARG DISTRO_NAME=apache-storm-${STORM_VERSION}
+
+ENV ZK_GPG_KEYS BBE7232D7991050B54C8EA0ADC08637CA615D22C
+
+ARG ZK_SHORT_DISTRO_NAME="zookeeper-3.8.0"
+ARG ZK_DISTRO_NAME="apache-zookeeper-3.8.0-bin"
+
+# Download Apache Storm and Apache Zookeeper
+RUN set -eux; \
+    ddist() { \
+        local f="$1"; shift; \
+        local distFile="$1"; shift; \
+        local success=; \
+        local distUrl=; \
+        for distUrl in \
+            'https://www.apache.org/dyn/closer.cgi?action=download&filename=' \
+            https://www-us.apache.org/dist/ \
+            https://www.apache.org/dist/ \
+            https://archive.apache.org/dist/ \
+        ; do \
+            if wget -q -O "$f" "$distUrl$distFile" && [ -s "$f" ]; then \
+                success=1; \
+                break; \
+            fi; \
+        done; \
+        [ -n "$success" ]; \
+    }; \
+    ddist "$DISTRO_NAME.tar.gz" "storm/$DISTRO_NAME/$DISTRO_NAME.tar.gz"; \
+    ddist "$DISTRO_NAME.tar.gz.asc" "storm/$DISTRO_NAME/$DISTRO_NAME.tar.gz.asc"; \
+    ddist "${ZK_DISTRO_NAME}.tar.gz" "zookeeper/${ZK_SHORT_DISTRO_NAME}/${ZK_DISTRO_NAME}.tar.gz"; \
+    ddist "${ZK_DISTRO_NAME}.tar.gz.asc" "zookeeper/${ZK_SHORT_DISTRO_NAME}/${ZK_DISTRO_NAME}.tar.gz.asc"; \
+
+#Verify Apache Storm PGP signature, untar and clean up.
+    export GNUPGHOME="$(mktemp -d)"; \
+    gpg --keyserver hkps://keyserver.pgp.com --recv-key "$GPG_KEY" || \
+    gpg --keyserver hkps://keyserver.ubuntu.com --recv-keys "$GPG_KEY" || \
+    gpg --keyserver hkps://pgp.mit.edu --recv-keys "$GPG_KEY"; \
+    gpg --batch --verify "$DISTRO_NAME.tar.gz.asc" "$DISTRO_NAME.tar.gz"; \
+    tar -xzf "$DISTRO_NAME.tar.gz"; \
+    rm -rf "$GNUPGHOME" "$DISTRO_NAME.tar.gz" "$DISTRO_NAME.tar.gz.asc"; \
+    chown -R storm:storm "$DISTRO_NAME" ; \
+
+#Verify Apache Zookeeper PGP signature, untar and clean up.
+    export GNUPGHOME="$(mktemp -d)"; \
+    gpg --keyserver ha.pool.sks-keyservers.net --recv-key ${ZK_GPG_KEYS} || \
+    gpg --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys ${ZK_GPG_KEYS} || \
+    gpg --keyserver keyserver.pgp.com --recv-keys ${ZK_GPG_KEYS}; \
+    gpg --batch --verify "${ZK_DISTRO_NAME}.tar.gz.asc" "${ZK_DISTRO_NAME}.tar.gz"; \
+    tar -zxf "${ZK_DISTRO_NAME}.tar.gz"; \
+    mv "${ZK_DISTRO_NAME}/conf/"* "${ZOO_CONF_DIR}"; \
+    rm -rf "${GNUPGHOME}" "${ZK_DISTRO_NAME}.tar.gz" "${ZK_DISTRO_NAME}.tar.gz.asc"; \
+    chown -R storm:storm "$ZK_DISTRO_NAME"
+
+
+WORKDIR $DISTRO_NAME
+
+ENV PATH $PATH:/$DISTRO_NAME/bin
+
+# ports
+EXPOSE 8080 8000
+
+# volume
+VOLUME ["/logs"]
+
+# copy configuration
+COPY zoo.cfg /apache-zookeeper-3.8.0-bin/conf/zoo.cfg
+
+COPY ./docker-entrypoint.sh /
+
+ENTRYPOINT ["/docker-entrypoint.sh"]
+
+CMD ["/apache-zookeeper-3.8.0-bin/bin/zkServer.sh", "start-foreground"]
```

Comment:
symlink or move folder. we can't rely on version number in folder name. it will generate more maintenance

Reply:
folder changed